src/gsi/cplr_read_wrf_mass_guess.f90

```diff
@@ -2240,6 +2297,74 @@ subroutine read_wrf_mass_netcdf_guess_wrf(this,mype)
                    end if
+
+! --- direct reflectivity DA --- CV transform on hydrometeors / CVq, CVlogq, and CVpq
+! Power transform for related q variables begins here==============================================! g.zhao
+!                  hydrometeors (reset zero cloud variables to tiny but ~0dbz)
+                  if(l_hydrometeor_bkio .and. n_actual_clouds>0) then
+                    ! =====================================================================!
+                     if (l_use_cvpqx) then
+                         if ( l_use_cvpqx_pval .eq. 0. ) then ! CVlogq
+                            if (ges_tsen(j,i,k,it) .gt. 274.15) then ! temperature dependent minimums
+                               qr_min=2.9E-6_r_kind
+                               qr_thrshd=qr_min * one_tenth
+                               qs_min=0.1E-9_r_kind
+                               qs_thrshd=qs_min
+                               qg_min=3.1E-7_r_kind
+                               qg_thrshd=qg_min * one_tenth
+                            else if (ges_tsen(j,i,k,it) .le. 274.15 .and. ges_tsen(j,i,k,it) .ge. 272.15) then
+                               qr_min=2.0E-6_r_kind
+                               qr_thrshd=qr_min * one_tenth
+                               qs_min=1.3E-7_r_kind
+                               qs_thrshd=qs_min * one_tenth
+                               qg_min=3.1E-7_r_kind
+                               qg_thrshd=qg_min * one_tenth
+                            else if (ges_tsen(j,i,k,it) .lt. 272.15) then
+                               qr_min=0.1E-9_r_kind
+                               qr_thrshd=qr_min
+                               qs_min=6.3E-6_r_kind
+                               qs_thrshd=qs_min * one_tenth
+                               qg_min=3.1E-7_r_kind
+                               qg_thrshd=qg_min * one_tenth
+                            end if
+                            if ( ges_qr(j,i,k) .le. qr_thrshd )  ges_qr(j,i,k) = qr_min
+                            if ( ges_qs(j,i,k) .le. qs_thrshd )  ges_qs(j,i,k) = qs_min
+                            if ( ges_qg(j,i,k) .le. qg_thrshd )  ges_qg(j,i,k) = qg_min
+
+!                           Logarithmic conversion of cloud hydrometer variables (qr,qs and qg)
+                            if(mype==0 .AND. i==3 .AND. j==3 .AND. k==3)then
+                                write(6,*)'read_wrf_mass_guess_netcdf: convert qr/qs/qg with log transform.'
+                            end if
+                            ges_qr(j,i,k) = log(ges_qr(j,i,k))
+                            ges_qs(j,i,k) = log(ges_qs(j,i,k))
+                            ges_qg(j,i,k) = log(ges_qg(j,i,k))
+                         else if ( l_use_cvpqx_pval .gt. 0. ) then   ! CVpq
+!                           convert cloud hydrometer variables (qr,qs and qg) to log
+                            if(mype==0 .AND. i==3 .AND. j==3 .AND. k==3)then
```

Comment:
see my comment on this kind of write command.

Reply:
I modified the code according to your comments. I took out these codes from loops and make them print one time at the start of the routine using if(mype==0).

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-9454084e44d25b1106008105c0d11d4efee5640bbe7134a7f8faa41509269918R2836-R2851